smaily_for_drupal/src/Form/SubscribeForm.php

```diff
@@ -52,87 +68,35 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $config = $this->config('smaily_for_drupal.settings');
-    $block_config = $form_state->getBuildInfo()['args'][0];
-
-    $username = $config->get('smaily_api_credentials.username');
-    $password = $config->get('smaily_api_credentials.password');
-    $domain = $config->get('smaily_api_credentials.domain');
-
-    $query_data = [
-      'autoresponder' => $block_config['autoresponder'],
-      'addresses' => [
-        [
-          'email' => $form_state->getValue('email'),
-          'name' => $form_state->getValue('name') ?: '',
-        ],
-      ],
-    ];
-    // Append selected custom options to query.
-    $categories = $form_state->getValue('custom_fields');
-    if ($categories) {
-      foreach ($categories as $key => $value) {
-        $query_data['addresses'][0][$key] = 1;
-      }
-    }
+    // Fullfill FormBase.
+  }
 
-    $ch = curl_init();
-    curl_setopt($ch, CURLOPT_URL, 'https://' . $domain . '.sendsmaily.net/api/autoresponder.php');
-    curl_setopt($ch, CURLOPT_POST, TRUE);
-    curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($query_data));
-    curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
-    curl_setopt($ch, CURLOPT_HEADER, FALSE);
-    curl_setopt($ch, CURLOPT_USERPWD, "$username:$password");
-    $response = json_decode(curl_exec($ch), TRUE);
-    $http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-    curl_close($ch);
-
-    if ($http_code === 200 && !empty($response)) {
-      if (array_key_exists('code', $response)) {
-        switch ((int) $response['code']) {
-          case 101:
-            $this->messenger()->addMessage($this->t('Successfully subscribed to mailing list.'));
-            return;
-
-          case 201:
-            $this->messenger()->addWarning($this->t('Data must be posted with POST method.'));
-            return;
-
-          case 204:
-            $this->messenger()->addWarning($this->t('Input does not contain a valid email address.'));
-            return;
-
-          case 205:
-            $this->messenger()->addWarning($this->t(
-              'Could not add to subscriber list for an unknown reason. Probably something in Smaily.'
-            ));
-            return;
-
-          default:
-            $this->messenger()->addError($this->t('Something went wrong.'));
-            $this->logger('smaily')->error(nl2br(print_r(
-              array_merge(['query_parameters' => $query_data], ['response' => (array) $response]), TRUE
-            )));
-            return;
-        }
-      }
-    }
-    $this->messenger()->addError($this->t('Something went wrong with connecting to Smaily.'));
-    $this->logger('smaily')->error(nl2br(print_r(
-      array_merge(['query_parameters' => $query_data], ['response' => (array) $response]), TRUE
-    )));
+  /**
+   * Remove form_id and form_build_id as to not send them to Smaily as fields.
+   */
+  public function cleanHiddenDrupalInputs($form) {
```

Comment:
Term "clean" does not reflect what is actually done. I would assume hidden Drupal inputs are somehow normalized (values changed, names updated, etc), but they would still exist. I think it is better to use "remove" here, because it sends a strong signal that hidden Drupal inputs are removed from the form.